.github/workflows/run-tests.yml

```diff
@@ -0,0 +1,65 @@
+name: ci testing
+
+
+on:
+  push:
+    branches: master
+  pull_request:
+
+
+jobs:
+
+  run_test:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
```

Comment:
Is there a rationale on not setting it to `true`?

Reply:
Default is indeed `true`, but this means that as soon as one of the cases of the matrix that are run in parallel (i.e. `python3.7`, `python3.8`, `python3.9`) fails, all the other ones are also cancelled and marked as "failed".

I think it makes more sense to see exactly _which_ tests are failing, it may happen (we saw that recently) that e.g. we fail `3.9` but pass `3.7` and `3.8` which is more insightful than just seeing everything as "failed".